autogpt_platform/frontend/src/components/agents/agent-run-draft-view.tsx

```diff
@@ -55,10 +57,12 @@ export default function AgentRunDraftView({
 
   const doRun = useCallback(
     () =>
-      api
-        .executeGraph(graph.id, graph.version, inputValues)
-        .then((newRun) => onRun(newRun.graph_exec_id)),
-    [api, graph, inputValues, onRun],
+      toastOnFail("execute agent", () =>
+        api
+          .executeGraph(graph.id, graph.version, inputValues)
+          .then((newRun) => onRun(newRun.graph_exec_id)),
+      ),
```

Comment:
Why the wrapping construction?
```suggestion
      api
        .executeGraph(graph.id, graph.version, inputValues)
          .then((newRun) => onRun(newRun.graph_exec_id)),
          .catch((error) => toastFailure("execute agent", error)),
```

Reply:
yeah that works too, you just need to change more code on the existing implementation.

Updated